syntaxes/GDShader.tmLanguage.json

```diff
@@ -77,7 +77,7 @@
         },
         "hintKeyword": {
             "name": "storage.type.gdshader",
-            "match": "\\bhint_(?:color|range|(?:black_)?albedo|normal|white|black|aniso)\\b"
+            "match": "\\bsource_color|hint_(?:default_)?(?:color|range|(?:black_)?albedo|normal|white|black|aniso)\\b"
```

Comment:
This regex allows things like `hint_default_aniso`, `hint_default_albedo`, etc. Is this desired? It seems it should be only `hint_default_black` and `hint_default_white` from what I understood.
Also `\b` should apply in both sides on all words, so `|` requires an extra `(?:)` surrounding it. I suggest:
`"\\b(?:source_color|hint_(?:color|range|(?:black_)?albedo|normal|(?:default_)?(?:white|black)|aniso))\\b"`

Reply:
Good catch, I switched to your suggestion.

> This regex allows things like hint_default_aniso, hint_default_albedo, etc. Is this desired?

Personally, I don't think it's desirable to hardcode the hint names in the first place: my instinct is to match based on the structure of the variable declaration. I also don't know shaders very well, does anything other than a hint ever appear after the colon?

(Just to be safe, I'm not interested in restructuring the grammar. I think the differences in our approaches are pretty interesting.)

> Also \b should apply in both sides on all words, so | requires an extra (?:) surrounding it.

I haven't used `\b`, so thanks for the information.

Reply:
> Personally, I don't think it's desirable to hardcode the hint names in the first place: my instinct is to match based on the structure of the variable declaration. I also don't know shaders very well, does anything other than a hint ever appear after the colon?

Yes, I agree that a generic rule for that is far better ... as long as those are not keywords, i.e. does trying to declare a variable with that name result in error? If so, it's better to list them, because they would be actually reserved words. I can't remember if that's the case.

Reply:
> I haven't used \b, so thanks for the information.
It's a word boundary assertion. It's useful for preventing things like it matching `int` in a word like `hint`.